compiler/src/iree/compiler/Codegen/Common/BlockDynamicDimensions.cpp

```diff
@@ -102,7 +102,11 @@ getTensorDivisibilityInfo(const TensorDynamicDimAnalysis &dynamicDimAnalysis,
 /// inverses of each other. The `util.optimization.barrier` avoid these from
 /// getting folded away during reshape propagation. Return the result of the
 /// `tensor.collapse_shape generated.
-static std::optional<Value>
+struct ReshapeOps {
+  tensor::ExpandShapeOp expandShapeOp;
+  tensor::CollapseShapeOp collapseShapeOp;
+};
```

Comment:
Can you also include the optimization barrier here? You can then delete the optimization barriers created by this transformation only, which was weird previously, since it deletes all optimization barriers.

Reply:
don't have to do it if it's too much work, just would be nice for completeness.

Reply:
I could, but really we should have any `util.optimization_barriers` in the kernel :D . Ill leave it as is for now, but should be easy to fix later.